src/dstack/_internal/server/migrations/versions/1338b788b612_reverse_job_instance_relationship.py

```diff
@@ -0,0 +1,69 @@
+"""Reverse Job-Instance relationship
+
+Revision ID: 1338b788b612
+Revises: da574e93fee0
+Create Date: 2025-01-16 14:59:19.113534
+
+"""
+
+import sqlalchemy as sa
+import sqlalchemy_utils
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "1338b788b612"
+down_revision = "da574e93fee0"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    with op.batch_alter_table("jobs", schema=None) as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "instance_id", sqlalchemy_utils.types.uuid.UUIDType(binary=False), nullable=True
+            )
+        )
+        batch_op.create_foreign_key(
+            batch_op.f("fk_jobs_instance_id_instances"),
+            "instances",
+            ["instance_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+
+    op.execute("""
+        UPDATE jobs AS j
+        SET instance_id = (
+            SELECT i.id
+            FROM instances AS i
+            WHERE i.job_id = j.id
+        )
+    """)
+
+    with op.batch_alter_table("instances", schema=None) as batch_op:
+        batch_op.drop_constraint("fk_instances_job_id_jobs", type_="foreignkey")
+        batch_op.drop_column("job_id")
+
+
+def downgrade() -> None:
+    with op.batch_alter_table("instances", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("job_id", sa.UUID(), autoincrement=False, nullable=True))
```

Comment:
sa.UUID() fails with sqlite

change to:

```
batch_op.add_column(sa.Column("job_id", sqlalchemy_utils.types.uuid.UUIDType(binary=False), nullable=True))
```

Reply:
https://github.com/dstackai/dstack/pull/2253/commits/a6dcde20a4a673cb0b458a24389cfecc68a058c7